components/GameDetailForm/GameDetailForm.js

```diff
@@ -101,196 +100,202 @@ export default function GameDetails({ id, showMoreDetails, results }) {
-          id="firstsetplayertwo"
-          aria-label="Set 1: Score player two"
-          value={firstSetScorePlayerTwo}
-          onChange={(event) => setFirstSetScorePlayerTwo(event.target.value)}
-        />
-      </Fieldsets>
+        <Fieldsets>
+          <SetLabel aria-label="Score of first set" htmlFor="firstset">
+            Set 1
+          </SetLabel>
+          <SetInput
+            role="numberinput"
+            type="number"
+            min="0"
+            max="20"
+            name="firstsetplayerone"
+            id="firstsetplayerone"
+            aria-label="Set 1: Score player one"
+            value={firstSetScorePlayerOne}
+            onChange={(event) => setFirstSetScorePlayerOne(event.target.value)}
+          />
+          <SetInput
+            role="numberinput"
+            type="number"
+            min="0"
+            max="20"
+            name="firstsetplayertwo"
+            id="firstsetplayertwo"
+            aria-label="Set 1: Score player two"
+            value={firstSetScorePlayerTwo}
+            onChange={(event) => setFirstSetScorePlayerTwo(event.target.value)}
+          />
+        </Fieldsets>
 
-      <Fieldsets>
-        <SetLabel aria-label="Score of second set" htmlFor="secondset">
-          Set 2
-        </SetLabel>
-        <SetInput
-          role="numberinput"
-          type="number"
-          min="0"
-          max="20"
-          name="secondsetplayerone"
-          id="secondsetplayerone"
-          aria-label="Set 2: Score player one"
-          value={secondSetScorePlayerOne}
-          onChange={(event) => setSecondSetScorePlayerOne(event.target.value)}
-        />
-        <SetInput
-          role="numberinput"
-          type="number"
-          min="0"
-          max="20"
-          name="secondsetplayertwo"
-          id="secondsetplayertwo"
-          aria-label="Set 2: Score player two"
-          value={secondSetScorePlayerTwo}
-          onChange={(event) => setSecondSetScorePlayerTwo(event.target.value)}
-        />
-      </Fieldsets>
+        <Fieldsets>
+          <SetLabel aria-label="Score of second set" htmlFor="secondset">
+            Set 2
+          </SetLabel>
+          <SetInput
+            role="numberinput"
+            type="number"
+            min="0"
+            max="20"
+            name="secondsetplayerone"
+            id="secondsetplayerone"
+            aria-label="Set 2: Score player one"
+            value={secondSetScorePlayerOne}
+            onChange={(event) => setSecondSetScorePlayerOne(event.target.value)}
+          />
+          <SetInput
+            role="numberinput"
+            type="number"
+            min="0"
+            max="20"
+            name="secondsetplayertwo"
+            id="secondsetplayertwo"
+            aria-label="Set 2: Score player two"
+            value={secondSetScorePlayerTwo}
+            onChange={(event) => setSecondSetScorePlayerTwo(event.target.value)}
+          />
+        </Fieldsets>
```

Comment:
as I read the function setValidation() above, I guess players are not allow to set further scores if the first one has not been set yet? I would just display further Fieldsets if the current score has set. Something like set1&&(Fieldsets/) (sori can not write tag here so have to replace with the round bracket). In this way u dont have to write a complicate check up mechanismm for the validation. But for now i guess u can just leave it like that..

Reply:
thanks will do that in the future :D